pkg/geth/node/node.go

```diff
@@ -5,45 +5,82 @@ import (
 
 	"strconv"
 
+	"regexp"
+
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/p2p"
 	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/vulcanize/vulcanizedb/pkg/core"
 )
 
-func Info(client *rpc.Client) core.Node {
+type Getter interface {
+	NetworkId() float64
+	GenesisBlock() string
+	GethNodeInfo() (string, string)
+	ParityNodeInfo() (string, string)
+}
+
+type NodeFactory struct {
+	Client *rpc.Client
+}
+
+func MakeNode(getter Getter) core.Node {
 	node := core.Node{}
-	node.NetworkID = NetworkID(client)
-	node.GenesisBlock = GenesisBlock(client)
-	node.ID, node.ClientName = IDClientName(client)
+	node.NetworkID = getter.NetworkId()
+	node.GenesisBlock = getter.GenesisBlock()
+	node.ID, node.ClientName = getter.GethNodeInfo()
+	if notGethNode(node) {
+		node.ID, node.ClientName = getter.ParityNodeInfo()
+	}
 	return node
 }
 
-func IDClientName(client *rpc.Client) (string, string) {
+func notGethNode(node core.Node) bool {
+	return node.ID == ""
```

Comment:
If we find ourselves writing functions to infer the type of the node more frequently, perhaps we want to consider exposing a property on the node that wraps all of that logic and just returns what type it is